service/share/discovery/discovery.go

```diff
@@ -0,0 +1,66 @@
+package discovery
+
+import (
+	"context"
+	"time"
+
+	core "github.com/libp2p/go-libp2p-core/discovery"
+	discovery "github.com/libp2p/go-libp2p-discovery"
+)
+
+const (
+	namespace = "full"
+	interval  = time.Second * 10
+)
+
+// Advertise is a utility function that persistently advertises a service through an Advertiser.
+func Advertise(ctx context.Context, a core.Advertiser) {
+	go func() {
+		for {
+			ttl, err := a.Advertise(ctx, namespace)
+			if err != nil {
+				log.Debugf("Error advertising %s: %s", namespace, err.Error())
+				if ctx.Err() != nil {
+					return
+				}
+
+				select {
+				case <-time.After(interval):
+					continue
+				case <-ctx.Done():
+					return
+				}
+			}
+
+			wait := 7 * ttl / 8
```

Comment:
Will it be better to declare `7` and `8` as const var at the top?

Reply:
added a calculation as a var in the top